### ISSUE TITLE
`susie_suff_stat `: Arguments changed?

### ISSUE THREAD

**Issue by bschilder** [stephenslab/susieR#169]
It seems the arguments to `susie_suff_stat ` recently changed.

Could you add argument-specific deprecation warnings, or make the function backcompatible with the previous arguments?

## Reprex

```R
library(susieR)
data(N3finemapping)
attach(N3finemapping)

sumstats <- univariate_regression(X, Y[,1])
z_scores <- sumstats$betahat / sumstats$sebetahat
R <- cor(X)

fitted_bhat <- susieR::susie_suff_stat(
    bhat = sumstats$betahat,
    shat = sumstats$sebetahat,
    # maf = if("MAF" %in% colnames(dat)) dat$MAF else NULL,
    R = R,
    n = 1474097, # Number of samples/individuals in the dataset
    L = 5 # maximum number of non-zero effects 
)

```

### Using `susieR` v0.12.16:

```
Error in susie_func(bhat = dat$Effect, shat = dat$StdErr, maf = if ("MAF" %in%  : 
  unused arguments (bhat = dat$Effect, shat = dat$StdErr, R = base::data.matrix(LD_matrix))
```

### Using `susieR` v0.11.92:

Runs without any error.

Best, 
Brian

**Comment by bschilder** [comment]
Hi @pcarbo, I actually `susie_rss` in other scenarios, but I still think `susie_suff_stat` should be backcompatilbe to avoid breaking code that relies on it. 

If there is a strong reason to change the argument names, best practice in any package is to provide deprecation notices for at least 1 major version release. Packages like `lifecycle` make facilitate this:
https://lifecycle.r-lib.org/reference/deprecate_soft.html

This came up because [`echofinemap`](https://github.com/RajLabMSSM/echofinemap) uses `susie_suff_stat` internally, and it wasn't clear from the errors why this function had suddenly stopped working.

**Comment by zouyuxin** [comment]
@pcarbo yes. In <12 version, `susie_suff_stat` takes either 1.(bhat, shat, R, n, var_y) or 2. (XtX, Xty, yty, n). We compute 2 if 1 is provided. In version 12, we decided to simplify the interface and moved the computation to `susie_rss`, so `susie_suff_stat` takes XtX, Xty, yty, n only, and susie_rss is the main interface.

**Comment by pcarbo** [comment]
I pushed a change to provide an informative message for those trying to use an earlier version of the interface:

```R
susie_suff_stat(bhat = 1,shat = 1,n = 100,R = 1)
# Error in susie_suff_stat(bhat = 1, shat = 1, n = 100, R = 1) :
# susie_suff_stat no longer accepts inputs bhat, shat, R or var_y; these inputs are now 
# accepted by susie_rss instead
```
